File: src/144.go
package main

import (
	"fmt"
)

func main() {
	t := TreeNode{1, nil, nil}
	t.Left = &TreeNode{2, nil, nil}
	t.Right = &TreeNode{3, nil, nil}
	var r []int
	r = preorderTraversal(&t)
	fmt.Println(r)
}

type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

func traversal(root *TreeNode, result []int) {
	result = append(result, root.Val)
	if root.Left != nil {
		traversal(root.Left, result)
	}
	if root.Right != nil {
		traversal(root.Right, result)
	}
}

func preorderTraversal(root *TreeNode) []int {
	var result []int
	traversal(root, result)
	return result
}

File: src/144.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def __init__(self):
        self.List = []
    def preorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        self.traversal(root)
        return self.List
    def traversal(self, root):
        if root == None:
            return
        else:
            self.List.append(root.val)
            self.traversal(root.left)
            self.traversal(root.right)

solution = Solution()
print(solution.preorderTraversal(None))

    